### PR TITLE
[`semicolon_if_nothing_returned`]: add an autofix

### PR DESCRIPTION
**Pull request by sylvestre**
changelog: [`semicolon_if_nothing_returned`]: add an autofix

### CODE REVIEW

File: clippy_lints/src/semicolon_if_nothing_returned.rs [hunk 1]
```diff
@@ -52,14 +52,19 @@ impl<'tcx> LateLintPass<'tcx> for SemicolonIfNothingReturned {
                 if let ExprKind::DropTemps(..) = &expr.kind {
                     return;
                 }
-                span_lint_and_sugg(
+                span_lint_and_then(
```
Comment (Centri3) [inline clippy_lints/src/semicolon_if_nothing_returned.rs:55]: This is actually unnecessary, you only need to change `app` to `MachineApplicable`.

Comment (sylvestre) [reply clippy_lints/src/semicolon_if_nothing_returned.rs]: oh, it is magic :)
thanks
pushed!

**Review by dswij — COMMENTED** [review]
Thank you.

Can you also help to add the test for autofix?

You'll need to add `//@run-rustfix` header at the top of `tests/ui/semicolon_if_nothing_returned.rs`.

A `.fixed` file will be generated that will result in the test failure. If nothing is weird, you'll need to run the test again with `--bless` to copy the generated file as the expected output.

**Review by dswij — APPROVED** [review]

**Comment by dswij** [comment]
> @dswij done, sorry it took that long :)

No worries at all. Thank you for taking the time to add the test!

@bors r+

**Comment by sylvestre** [comment]
@dswij
I guess it failed because it was
changelog:
fxx

I updated the initial PR comment to be on only one line

File: tests/ui/semicolon_if_nothing_returned.fixed [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    repro!();
+}
+
+fn function_returning_option() -> Option<i32> {
+    Some(1)
+}
+
+// No warning
+fn let_else_stmts() {
+    let Some(x) = function_returning_option() else { return; };
```
Comment (Centri3) [inline tests/ui/semicolon_if_nothing_returned.fixed:120]: Seems you'll need to run `cargo dev fmt`.

`let-else` is now, (finally!), formatted. That seems to be the only issue.

Comment (sylvestre) [reply tests/ui/semicolon_if_nothing_returned.fixed:120]: @Centri3 
are you sure? cargo dev fmt doesn't make any change on my system
with 
rustfmt 1.5.3-nightly (5bd28f5 2023-06-28)

Comment (Centri3) [reply tests/ui/semicolon_if_nothing_returned.fixed:120]: 2023-06-28 is pretty old, current `rust-toolchain` shows `2023-07-14` so you'll likely want to `git pull upstream master`

Comment (sylvestre) [reply tests/ui/semicolon_if_nothing_returned.fixed]: my bad, rustup didn't updated to this one ?!
should be good now!

**Comment by sylvestre** [comment]
@bors retry
(in case I have permissions)

**Comment by dswij** [comment]
@sylvestre Thank you for this and @Centri3 for helping

Apologies for the slow response at times and the hurdles in merging the PR 😕

**Comment by sylvestre** [comment]
ouhlala, don't apologize :) this was perfect!

**Pull request merged**